pkg/apis/resource/validation/validation.go

```diff
@@ -372,18 +393,29 @@ func validateResourceClaimStatusUpdate(status, oldStatus *resource.ResourceClaim
 		func(consumer resource.ResourceClaimConsumerReference) (types.UID, string) { return consumer.UID, "uid" },
 		fldPath.Child("reservedFor"))...)
 
-	var allocatedDevices sets.Set[structured.DeviceID]
+	var allocatedDevices sets.Set[structured.SharedDeviceID]
 	if status.Allocation != nil {
 		allocatedDevices = gatherAllocatedDevices(&status.Allocation.Devices)
 	}
-	allErrs = append(allErrs, validateSet(status.Devices, -1,
-		func(device resource.AllocatedDeviceStatus, fldPath *field.Path) field.ErrorList {
-			return validateDeviceStatus(device, fldPath, allocatedDevices)
-		},
-		func(device resource.AllocatedDeviceStatus) (structured.DeviceID, string) {
-			return structured.MakeDeviceID(device.Driver, device.Pool, device.Device), "deviceID"
-		},
-		fldPath.Child("devices"))...)
+	validateItem := func(device resource.AllocatedDeviceStatus, fldPath *field.Path) field.ErrorList {
+		return validateDeviceStatus(device, fldPath, allocatedDevices)
+	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRAConsumableCapacity) || shareIDInUse(status) {
```

Comment:
don't check feature gates in validation, validate whatever data is actually in the object

```suggestion
	if shareIDInUse(status) {
```

is there a reason not to always use the shared device id validation path? when shareID is nil, that still works properly, right?

Reply:
According to [this change](https://github.com/kubernetes/kubernetes/pull/132522#discussion_r2241287958), this was also removed.